### PR TITLE
Workaround for weird linker issue with connect/qtw on windows

### PR DESCRIPTION
**Pull request by oyvindln**
Using connect with scaleDivChanged gives a weird linker error on windows for some reason. I could not figure out why, but using the old-style connect that operates on ```const char*``` works fine.

Add math define to make M_PI available on windows (M_PI is not-standard but a common extension, there is no built-in pi constant in std c++ headers before c++20)

also bump pro files to c++17. C++11 is ancient, c++17 is the default on the system we support I believe.

### CODE REVIEW

File: tools/ld-analyse/ld-analyse.pro [hunk 1]
```diff
@@ -159,6 +160,7 @@ unix:!macx {
 win32: {
     LIBS += qwt.lib
 }
+win32:INCLUDEPATH += "D:\prog\vcpkg\installed\x64-windows\include\qwt"
```
Comment (atsampson) [inline tools/ld-analyse/ld-analyse.pro:163]: Is that addition intentional? I assume that's a path on your machine...

**Comment by atsampson** [comment]
This looks basically OK to me. If we're requiring C++17, then there are some other cleanups that I can do as well (removing some C++14 workarounds, and I think there were a couple of places that could use `make_unique`).

With the connect calls, could you put those in a conditional for Windows, so other OSs don't lose the benefit of type-safety?

File: tools/ld-analyse/ld-analyse.pro (hunk 1 above)
Comment (oyvindln) [reply tools/ld-analyse/ld-analyse.pro:163]: Ah yeah that  just me forgetting to remove it

**Comment by oyvindln** [comment]
Okay, can make it conditional and leave the type safe connects in for non-windows.

Are there any possible systems we still support where C++17 support would be missing? Even Ubuntu 18.04 has GCC 7.3 (and optionallly gcc 8) which supports most of c++17 other than a few lesser used bits, though I think we might require 20.04 as minimum now anyhow which has GCC 9.3 so it shouldn't be any issues using that as the default and not needing workarounds for older ones.

**Comment by atsampson** [comment]
I don't think so - the CI tests use Ubuntu 20.04, which is the oldest version we support, and it's building OK there.

**Comment by atsampson** [comment]
Looks good. I'll merge these by hand along with removing the pre-C++17 workarounds.
